x/gamm/keeper/pool_service_test.go

```diff
@@ -840,29 +842,46 @@ func (suite *KeeperTestSuite) TestJoinSwapExactAmountInConsistency() {
 		shareOutMinAmount sdk.Int
 		expectedSharesOut sdk.Int
 		tokenOutMinAmount sdk.Int
+		expectError       error
 	}{
 		{
-			name:              "single coin with zero swap and exit fees",
+			name:              "happy path: single coin with zero swap and exit fees",
 			poolSwapFee:       sdk.ZeroDec(),
 			poolExitFee:       sdk.ZeroDec(),
-			tokensIn:          sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(1000000))),
+			tokensIn:          sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(1_000_000))),
 			shareOutMinAmount: sdk.ZeroInt(),
 			expectedSharesOut: sdk.NewInt(6265857020099440400),
 			tokenOutMinAmount: sdk.ZeroInt(),
 		},
-		// TODO: Uncomment or remove this following test case once the referenced
-		// issue is resolved.
-		//
-		// Ref: https://github.com/osmosis-labs/osmosis/issues/1196
-		// {
-		// 	name:              "single coin with positive swap fee and zero exit fee",
-		// 	poolSwapFee:       sdk.NewDecWithPrec(1, 2),
-		// 	poolExitFee:       sdk.ZeroDec(),
-		// 	tokensIn:          sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(1000000))),
-		// 	shareOutMinAmount: sdk.ZeroInt(),
-		// 	expectedSharesOut: sdk.NewInt(6226484702880621000),
-		// 	tokenOutMinAmount: sdk.ZeroInt(),
-		// },
+		{
+			name:              "corner case: single coin with positive swap fee and zero exit fee",
+			poolSwapFee:       sdk.NewDecWithPrec(1, 2),
+			poolExitFee:       sdk.ZeroDec(),
+			tokensIn:          sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(1_000_000))),
+			shareOutMinAmount: sdk.ZeroInt(),
+			expectedSharesOut: sdk.NewInt(6226484702880621000),
```

Comment:
I think it would be nice if we had a link to the calculations, which led to this number. Reference: https://github.com/osmosis-labs/osmosis/blob/main/osmomath/decimal_test.go#L724

Reply:
It might be out of scope of this PR, but I think it is not a big deal if we add it in it, if people agree with the importance of it